ui/js/directives/treeherder/clonejobs.js

```diff
@@ -196,10 +200,24 @@ treeherder.directive('thCloneJobs', [
             }
         };
 
+        var clickRunnableJobCb = function(el, resultset_id) {
+            var buildername = el.attr(runnableJobBuildernameAttr);
+            ThResultSetStore.toggleSelectedRunnableJob($rootScope.repoName, resultset_id, buildername);
+            el.toggleClass("runnable-job-btn-selected");
+        };
+
         var togglePinJobCb = function(ev, el, job){
             $rootScope.$emit(thEvents.jobPin, job);
         };
 
+        var filterWithRunnable = function(job) {
+            var rsMap = ThResultSetStore.getResultSetsMap($rootScope.repoName);
```

Comment:
I would move this get of ``rsMap`` to inside the ``if`` statement, so you don't waste time assigning if the job isn't ``runnable``.